### DISCUSSION TITLE
Phased Exposure Greyed Out

### DISCUSSION THREAD

**Discussion by JGarrett7** [sn4k3/UVtools#888] [Q&A] [answered]
### System

```Text
UVtools v4.3.2 X64
Operative system: Microsoft Windows 10.0.22631 X64
Processor: AMD Ryzen 7 7735U with Radeon Graphics         
Processor cores: 16
Memory RAM: 11.77 / 13.69 GB
Runtime: win10-x64
Framework: .NET 6.0.29
AvaloniaUI: 11.0.10
OpenCV: 4.8.1

Sreens, resolution, working area, usable area:
1: 1920 x 1200 @ 125% (Primary) (On this)
    WA: 1920 x 1140    UA: 1536 x 912

Path:       C:\Program Files\UVtools\
Executable: C:\Program Files\UVtools\UVtools.exe
Loaded file: StretchMagicMoldHolder v4.pwmo [Version: 515] [Class: PhotonWorkshopFile]
```

### Printer and Slicer

- Printer: Anycubic Photon Mono
- Slicer: Photon Workshop

### Description of the bug

I've been using Double Exposure for a while now. I noticed in the change log that it has been moved to the "phased exposure" tool, but that tool is greyed out for me. I double-checked with some old files to make sure it wasn't an issue with the new Photon Workshop.
![image](https://github.com/sn4k3/UVtools/assets/43509109/ccf5d491-d6f5-4863-afea-67033a1201f1)

### How to reproduce

Open a pwmo file and try to click "phased exposure"

### Files

[StretchMagicMoldHolder v4.zip](https://github.com/user-attachments/files/15875976/StretchMagicMoldHolder.v4.zip)

**Comment by sn4k3** [comment]
Any grey-out feature means not compatible with your printer and or file format.
Anycubic lacks of layers in same position, every layer must have unique height, as so phased exposure is disabled for you.
If old double exposure works was due the lack of that that safe-check in UVtools, making a offset in height.

**Reply by JGarrett7** [reply]
Odd, I've printed a lot with it and had great results. I successfully fixed probably 10 or so elephant foot issues using this technique. The dimensions on those parts were also correct, so I don't know what it was doing then.

**Reply by sn4k3** [reply]
Do you have an old file using the double exposure method you can share?

**Reply by JGarrett7** [reply]
Here is one. I know the RERF file that Anycubic put out with my printer does multiple exposure lengths on one layer. Not quite the same thing, but similar (https://www.fabbaloo.com/news/what-is-an-r_e_r_f-
[_Head_UVTools.zip](https://github.com/user-attachments/files/15876952/_Head_UVTools.zip)
file#:~:text=Anycubic%20implemented%20a%20very%20unusual,GCODE%20file%2C%20%E2%80%9CR_E_F_R%E2%80%9D.).

**Reply by sn4k3** [reply]
I require a file you had actually printed with double exposure method applied.

**Reply by JGarrett7** [reply]
That should be the file I actually printed that removed the elephant's foot using the double exposure method. I just opened it again and it shows eroded perimeters every other layer.

**Reply by sn4k3** [reply]
But where's the file? I can't find any, fix the link or give a direct link to it

**Reply by JGarrett7** [reply]
I downloaded it from my message. It's the link reading [_Head_UVTools.zip](https://github.com/user-attachments/files/15876952/_Head_UVTools.zip)

**Reply by sn4k3** [reply]
I was getting redirected to "fabbaloo.com/...".
I open your file and it have the double exposures applied correctly. However I'm a bit skeptical about it actually run correctly, because so far no one had able to run "Multiple exposure" tests using anycubic machines as printer do not respect layer position with a "0mm" offset.

Can I send you a test file to print a cube of 10mm but all layer print at 0.10mm position? If machine does not respect position I expect cube to print with 10mm heigth, otherwise 0.10mm

**Reply by JGarrett7** [reply]
That sounds good. I am dealing with some dimensional inaccuracy on unmodified files in Z right now, but not to the level of 9.9 mm. I am happy to run the test.

**Reply by sn4k3** [reply]
Try this file: [cube_test.zip](https://github.com/user-attachments/files/15892277/cube_test.zip)
After 0.4mm all layers are printed in same height (0.4mm)
If printer is unable it will print cube up to 10mm

**Reply by JGarrett7** [reply]
You are right, it printed to 10mm. I guess the number of initial layers has been low enough that I haven't noticed the difference in height. Is there a way to accomplish the same thing by turning on pixels mid-layer like the RERF file does?

**Reply by sn4k3** [reply, marked as answer]
You can use pixel dimming or erode method

**Reply by JGarrett7** [reply]
Thanks so much! Sorry to bother you.
